PythonRpcServer/phrasehinter.py

```diff
@@ -94,40 +130,59 @@ def filter_common_corpus_words(words_count, scale_factor=300):
         corpus_freq = corpus_counts.get(word.lower(), 0) / corpus_total
         if word_freq >= (corpus_freq * scale_factor):
              result.append(word)
-
+    print(f"filter_common_corpus_words() complete. {len(result)} words in result")
     return result
 
 def require_minimum_occurence(transactions, min_support, abort_threshold=5000, maximum_phrase=500):
     """
     A function that extracts the mximal frequent sequential patterns from the raw string
     """
-
+    print(f"require_minimum_occurence() starting. {len(transactions)} transactions")
     # generate frequent sequential patterns through PrefixSpan library
     if len(transactions) > abort_threshold: # Return an empty result If N is too large
+        print("Too many transactions; returning empty list")
         return []
 
+    # Determine prefix span
+    # see https://pypi.org/project/prefixspan/
+    # The above docs suggest faster alternatives for large N
+    # Todo: investigate these
+    print("PrefixSpan starting") # This is likely the slowest part of processing
+    # Todo: As an alternative to returning an empty set 
+    # we could randomly sample the transactions e.g take abort_threshold samples
+    # Todo: put a time constraint on this?
+    start_time = perf_counter()
     ps = PrefixSpan(transactions)
+    end_time = perf_counter()
+    print(f"PrefixSpan complete. {int(end_time-start_time)} seconds")
+
+    print("prefixspan.frequent(...) starting")
+    start_time = perf_counter()
     pattern_count = ps.frequent(min_support, closed=True)
```

Comment:
From the documentation, it sounds like this `closed=True` is all that is needed to use the BIDE implementation, which they claim is most performant for large N. I'm still unclear if there are scenarios where BIDE is less performant than FEAT or the default PrefixSpan